src/app/designs/components/design-modules/design-module/design-module.component.ts

```diff
@@ -0,0 +1,192 @@
+import { ConfirmDialogService } from 'src/app/sei-cwd-common/confirm-dialog/service/confirm-dialog.service';
+
+@Component({
+  selector: 'cas-design-module',
+  templateUrl: './design-module.component.html',
+  styleUrls: ['./design-module.component.scss'],
+  changeDetection: ChangeDetectionStrategy.OnPush,
+})
+export class DesignModuleComponent implements OnInit, OnChanges {
+  @Input()
+  public designModule: DesignModule;
+
+  @Input()
+  public module: Module;
+
+  public isEditing = false;
+  public showOutputs = false;
+
+  private loadingSubject = new BehaviorSubject(false);
+  public loading$ = this.loadingSubject.asObservable();
+
+  private set loading(val: boolean) {
+    this.loadingSubject.next(val);
+  }
+
+  public variables$: Observable<Variable[]>;
+
+  private outputsSubject = new BehaviorSubject<ModuleOutput[]>([]);
+  public outputs$ = this.outputsSubject.asObservable();
+
+  public designOutputs$;
+
+  private savingSubject = new BehaviorSubject(false);
+  public saving$ = this.savingSubject.asObservable();
+
+  private set saving(val: boolean) {
+    this.savingSubject.next(val);
+  }
+
+  constructor(
+    private confirmationService: ConfirmDialogService,
+    private designModuleService: DesignModuleService,
+    private moduleService: ModuleService,
+    private variableQuery: VariablesQuery,
+    private designModuleQuery: DesignModuleQuery
+  ) {}
+
+  ngOnInit(): void {
+    this.variables$ = this.variableQuery.selectByDesignId(
+      this.designModule.designId
+    );
+
+    this.designOutputs$ = this.designModuleQuery.getOutputsByDesignId(
+      this.designModule.designId,
+      this.designModule.id
+    );
+  }
+
+  ngOnChanges() {
+    if (
+      this.module &&
+      this.module.versions &&
+      this.designModule.moduleVersion
+    ) {
+      const version = this.module.versions.find(
+        (x) => x.name == this.designModule.moduleVersion
+      );
+
+      if (version) {
+        this.outputsSubject.next(version.outputs);
+      }
+    }
+  }
+
+  delete() {
+    this.confirmationService
+      .confirmDialog(
+        'Delete Module',
+        `Are you sure you want to delete ${this.designModule.name}?`
+      )
+      .pipe(take(1))
+      .subscribe((result) => {
+        if (!result[this.confirmationService.WAS_CANCELLED]) {
+          this.designModuleService.delete(this.designModule.id).subscribe();
+        }
+      });
+  }
+
+  edit() {
+    this.showOutputs = false;
+    this.isEditing = !this.isEditing;
+
+    if (this.isEditing) {
+      // wait on first load, otherwise refresh in background
+      // in case signalR missed an update
+      if (!this.designModule.values || !this.module.versions) {
+        this.loading = true;
+      }
+
+      forkJoin({
+        designModule: this.designModuleService.load(this.designModule.id),
+        module: this.moduleService.loadModuleById(this.module.id),
+      }).subscribe(() => {
+        this.loading = false;
+      });
```

Comment:
Missing takeUntil(this.unsubscribe$)

Reply:
I don't think you need to unsubscribe from a forkJoin anyway, since it only fires when all of the inner observables are complete, and subscriptions are automatically unsubscribed on completed observables. 

refs:
https://stackoverflow.com/questions/40429666/how-to-unsubscribe-from-an-observable-returned-by-forkjoin